File: unit18/python-ds-practice/34_same_frequency/same_frequency.py
def same_frequency(num1, num2):
    """Do these nums have same frequencies of digits?
    
        >>> same_frequency(551122, 221515)
        True
        
        >>> same_frequency(321142, 3212215)
        False
        
        >>> same_frequency(1212, 2211)
        True
    """
    str1 = str(num1)
    str2 = str(num2)

    count1 = set([str1.count(char) for char in str1])
    count2 = set([str2.count(char) for char in str2])

    return count1 == count2
    

File: unit48/prob-freq-count-pointers/same-frequency.js
function freqCounter(str) {
  const obj = {}
  
  for (let char of str) {
    obj[char] = (obj[char] + 1) || 1
  }

  return obj
}

function sameFrequency(num1, num2) {
  const str1 = String(num1)
  const str2 = String(num2)

  if (str1.length !== str2.length) return false
  
  const str1Freq = freqCounter(str1)
  const str2Freq = freqCounter(str2)
  
  for (let char in str1Freq) {
    if (str1Freq[char] !== str2Freq[char]) return false
  }

  return true
}